roles/galaxy_post_install/tasks/signing_service.yml

```
@@ -0,0 +1,59 @@
+---
+- name: Set permissions on galaxy signing service gpg key file
+  file:
+    path: "{{ pulp_certs_dir }}/galaxy_signing_service.gpg"
+    mode: 0600
+    owner: "{{ pulp_user }}"
+    group: "{{ pulp_group }}"
+  become: true
+
+# FIXME: separate logic to create directory, and to customize its path
+- name: Set permissions on galaxy signing service script
+  file:
+    path: /var/lib/pulp/scripts/collection_sign.sh
+    mode: 0750
+    owner: "{{ pulp_user }}"
+    group: "{{ pulp_group }}"
+  become: true
+
+- name: Commands run as pulp_user
+  block:
+    - name: Inspect galaxy signing service GPG key file
+      shell: gpg --list-secret-keys < /etc/pulp/certs/galaxy_signing_service.gpg
+      register: galaxy_signing_key
+      changed_when: false
+      check_mode: false
+
+    - name: Import galaxy signing service GPG key file
+      command: gpg --batch --import /etc/pulp/certs/galaxy_signing_service.gpg
+      register: result
+      # Imported successfully: rc 0
+      # Already imported: rc 2
+      # If you try to import multiple keys but only some need to be imported: rc 2
+      changed_when: result.stderr is search("imported:")
+      failed_when: result.rc not in [0,2]
+
+    - debug:
+        var: galaxy_signing_key.stdout
+    - debug:
+        msg: >
+          {{ galaxy_signing_key.stdout | regex_search( '/([A-F0-9]{8})' , '\\1', multiline=True) }}
+
+    - name: Trust the galaxy signing service GPG key
+      shell: >
+        (echo trust &echo 5 &echo y &echo quit &echo save) | gpg --batch --command-fd 0 --edit-key
+        {{ galaxy_signing_key.stdout | regex_search( '/([A-F0-9]{8})' , '\\1', multiline=True) }}
```

given the output i see from `gpg --list-secret-keys` i think it should be
```suggestion
        {{ galaxy_signing_key.stdout | regex_search( '^\s*([A-F0-9]{40})\s*$' , '\\1', multiline=True) }}
```

---

That doesn't work for CentOS 7 though:

---

what about `gpg --list-secret-keys --fingerprint`?

---

So the design I have right now is to read the key before it is imported, not after.

I test it by creating the keys, export them, and then deleting the entire ~/.gnupg folder.

That command only outputs after it is imported.

But fear not, I have it working. I am using the 16 hexadecimal long key ID. Which seems to be suitable for every purpose I have run into.

---

or even `gpg --list-secret-keys --list-options show-only-fpr-mbox` (not sure if that command is old enough)

---

It's not present on EL7.

Thank you though!

---

I decided to use the 40 character long fingerprint after all, since I needed to change the "trust" command to a different subcommand that requires it.